server/aesdsocket.c

```diff
@@ -0,0 +1,217 @@
+/** Header files **/
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+#include <stdbool.h>
+#include <signal.h>
+#include <unistd.h>
+#include <fcntl.h>
+#include <netdb.h>
+#include <syslog.h>
+#include <errno.h>
+#include <sys/stat.h>
+#include <sys/socket.h>
+#include <sys/types.h>
+#include <arpa/inet.h>
+
+/** Macros **/
+#define FILE "/var/tmp/aesdsocketdata"
+#define PORT 9000
+#define MAX_BUFFER_SIZE (1024)
+
+/** Global variable defination **/
+int file_fd = 0;
+int socket_fd = 0;
+int connection = 0;
+
+
+void signal_handler(int signo)
+{
+	if (signo == SIGINT || signo == SIGTERM)
+	{
+		syslog(LOG_DEBUG, "Caught the signal, exiting...");
+		unlink(FILE); 
+		close(connection);
+		close(socket_fd);
+		closelog();
+	}
+	exit(EXIT_SUCCESS);
```

Comment:
@AshwiniCUB is this all safe to do in a signal handler?  Consider a design where you can handle cleanup outside the signal handler instead.